test/unit/resources/aws_rds_instance_test.rb

```diff
@@ -0,0 +1,83 @@
+require 'ostruct'
+require 'helper'
+require 'aws_rds_instance'
+
+# MRDSIB = MockRDSInstanceBackend
+# Abbreviation not used outside this file
+
+#=============================================================================#
+#                            Constructor Tests
+#=============================================================================#
+class AwsMDBIConstructor < Minitest::Test
+  def setup
+    AwsRdsInstance::BackendFactory.select(AwsMRDSIB::Empty)
+  end
+  
+  def test_constructor_no_args_raises
+    assert_raises(ArgumentError) { AwsRdsInstance.new }
+  end
+
+  def test_constructor_accept_scalar_param
+    AwsRdsInstance.new('test-instance-id')
+  end
+
+  def test_constructor_expected_well_formed_args
```

Comment:
This test should only be used when there are combinations of parameters that are valid (and other combinations that are not).  Here, we have a simple scalar ID pattern; I don't think this is needed.